Forge.TreeWalker.UnitTests/test/TreeSchemaValidatorTests.cs

```diff
@@ -0,0 +1,208 @@
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaInStringWithCustomRules()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(tardigradeSchemaString, stringRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaInStringWithCustomRulesInJSchema()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(tardigradeSchemaString, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaFromPathWithCustomRulesInString()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromPath(tardigradeSchemaPath, stringRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaFromPathWithCustomRulesInJSchema()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromPath(tardigradeSchemaPath, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemasFromDirectoryListWithCustomRulesInString()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromDirectory(schemaDirectoryPath, stringRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemasFromDirectoryListWithCustomRulesInJSchema()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromDirectory(schemaDirectoryPath, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res); 
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateInvalidSchemaInString()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(tardigradeSchemaString, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateInvalidSchemaDirectoryPath()
+        {
+            try
+            {
+                ForgeSchemaValidator.ValidateSchemaFromDirectory(invalidSchemaDirectoryPath, jschemaRules, false, out IList<ValidationError> errorList);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual("The directory name is invalid.\r\n", e.Message);
+            }
```

Comment:
If your test is verifying an exception gets thrown, you need to make sure your UT hits some Assert. In this case, what if ValidateSchemaFromDirectory was successful? It wouldn't hit the catch block and the UT passes unexpectedly.

You can accomplish this in multiple ways. One is to use Assert.ThrowsException. Another way is to set a flag in the catch, then confirm that flag was set after the try/catch.